okhttp-ws/src/main/java/okhttp3/internal/ws/RealWebSocket.java

```diff
@@ -166,6 +169,9 @@ public void sendPong(Buffer payload) throws IOException {
 
     try {
       writer.writeClose(code, reason);
+
+      // Wait graciously 5 seconds for server close response before shutdown
+      this.source.timeout().timeout(5, TimeUnit.SECONDS);
```

Comment:
Probably nothing terrible, although I wouldn't be surprised if someone eventually complained.